source/lubeck.d

```diff
@@ -1960,6 +1960,94 @@ struct QRResult(T)
     {
         return qrSolve!(allowDestroy)(matrix, tau, b);
     }
+    
+    /++
+    Extract the Q matrix
+    +/
+    auto Q(Flag!"allowDestroy" allowDestroy = No.allowDestroy)
+    {
+        auto work = [T.sizeof * matrix.length].uninitSlice!T;
```

Comment:
does `T.sizeof` is required here?

Reply:
I'll be honest, I really don't know. I had just copied the work variable from qrDecomp. I removed it and it still compiles...

Reply:
Looks like both are ok. The ideal way is to make a `lwork` query, some mir-lapack functions has workspace size quiery versions like `sysv_rook_wk`, but we didn't yet implemented it for geqrf.

The current version with fixed n can be merged as well.

http://scc.ustc.edu.cn/zlsc/sugon/intel/mkl/mkl_manual/GUID-8E285C9D-50B5-4ADA-8ADA-35FA32FF201A.htm

lda
INTEGER. The leading dimension of a; at least max(1, m).

lwork
INTEGER. The size of the work array (lwork ≥ n).

If lwork = -1, then a workspace query is assumed; the routine only calculates the optimal size of the work array, returns this value as the first entry of the work array, and no error message related to lwork is issued by xerbla.

See Application Notes for the suggested value of lwork.